keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/multiple/ModelVertex.java

```diff
@@ -0,0 +1,45 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.multiple;
+
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.NonProbabilistic;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.bool.BoolVertex;
+import io.improbable.keanu.vertices.bool.nonprobabilistic.operators.BoolModelResultVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.DoubleModelResultVertex;
+import io.improbable.keanu.vertices.intgr.IntegerVertex;
+import io.improbable.keanu.vertices.intgr.nonprobabilistic.operators.IntegerModelResultVertex;
+
+import java.util.Map;
+
+public interface ModelVertex<T> extends NonProbabilistic<T> {
+
+    void run(Map<VertexLabel, Vertex<? extends Tensor>> inputs);
+
+    Map<VertexLabel, Tensor> updateValues(Map<VertexLabel, Vertex<? extends Tensor>> inputs);
+
+    boolean hasCalculated();
+
+    DoubleTensor getDoubleModelOutputValue(VertexLabel label);
+
+    IntegerTensor getIntegerModelOutputValue(VertexLabel label);
+
+    BooleanTensor getBooleanModelOutputValue(VertexLabel label);
```

Comment:
I don't think we should have these at all really - we'd be better off IMO just throwing an exception if anyone tries to get the value of a ModelVertex explicitly (as opposed to via an output Vertex).  We should also likely throw an exception if anyone tries to explicitly set the model vertex as a parent.

Reply:
Yeah - the algorithms will ask it to recalculate but the value will not be looked at. Throwing an exception will cause it to halt

Reply:
As mentioned above - these are needed for the current `ModelResultVertex` implementation